### PR TITLE
Support widescreen display.

### LINKED ISSUE #776: UI Components overlap on ultra-wide display
## Technical information
**Using version:**
 - [ ] `master` (running from GitHub-published source code, currently `v3.0.0-pre`)
 - [x] `latest` (latest release, currently `v2.2.2`)
 - [ ] `vX.X.X` (specify other version)

**Running on:**
 - [ ] Linux
 - [ ] Windows
 - [x] macOS

**How comfortable you are with your system and/or IT in general:**
 - [ ] I'm kind of lost, honestly
 - [ ] I know what's up, I could help you run some commands or checks
 - [x] My machine is fully under my control, tell me what you need
 - [ ] I attended [Defcon](https://defcon.org/) last year

---

## Problem

On 21:9 ultra-wide display, the network monitor overlaps with the keyboard. This effects all the build-in theme with keyboard.

Screen Shot 2020-09-29 at 10 55 07 PM

### PR DESCRIPTION
**Pull request by geosp**
Fixes #776

- [x] Fix overlap issues in multiple common display resolutions.
- [x] Fix setting modal panel by introducing scrolling in the settings table. 

Widescreen layout support:

widescreen-support

Fix setting modal:

settings-modal

No-type theme:

tron-notype

### CODE REVIEW

**Comment by geosp** [comment]
I have some additional settings that I think are useful, but since there is an issue with the settings dialog, I will wait until when or if this PR is approved to create another PR for them.

**Comment by geosp** [comment]
It will be really cool if we could explore migrating to using react with electron to build the UI.  Having a framework to create components will be awesome. Here is an example of a [layout manager](https://strml.github.io/react-grid-layout/examples/6-dynamic-add-remove.html). In this way, we can develop different components that can be dropped-in based on user preferences. I willing to help to make it happen.

**Comment by chelsey8** [comment]
okay

**Review by chelsey8 — APPROVED** [review]

**Review by chelsey8 — APPROVED** [review]

**Comment by GitSquared** [comment]
@geosp Sorry for the spam. I've contacted GitHub support as it seems this repo has been targeted lately.

> I have some additional settings that I think are useful, but since there is an issue with the settings dialog, I will wait until when or if this PR is approved to create another PR for them.

You should backport 17dc374 which includes a fix for this! The commit just before this one also contains dependency upgrades which should fix the failed test on your PRs.

> It will be really cool if we could explore migrating to using react with electron to build the UI. Having a framework to create components will be awesome. Here is an example of a layout manager. In this way, we can develop different components that can be dropped-in based on user preferences. I willing to help to make it happen.

I completely agree, and as #769 mentions, I would like to eventually rewrite the app in a modern tech stack. Unfortunately this would mean a lot of work and a time investment that I cannot afford right now. But one day!..

File: src/assets/css/modal.css [hunk 1]
```diff
@@ -132,6 +132,11 @@ div.modal_popup table:not(#settingsEditor) td:first-child {
     text-align: center;
 }
 
+div#settingsEditorContainer {
```
Comment (GitSquared) [inline src/assets/css/modal.css:135]: Sorry, didn't realize that you already fixed this. You're going to have to figure out with the merger with upstream...

File: src/assets/css/filesystem.css [hunk 2]
```diff
@@ -1,7 +1,7 @@
 section#filesystem {
     position: relative;
     top: -0.925vh;
-    width: 43vw;
+    width: 30vw;
```
Comment (GitSquared) [inline src/assets/css/filesystem.css:4]: Correct me if I'm wrong, but adapting the UI for larger screen ratios by changing the values like that should break it on more common screens, which is not something we want!

You should apply your changes by overriding the CSS in a media query rule, in [this file](https://github.com/GitSquared/edex-ui/blob/master/src/assets/css/extra_ratios.css).

Just copy your changes there in the correct media rule, and it should be fine.

File: src/assets/themes/apollo-notype.json [hunk 3]
```diff
@@ -26,5 +26,5 @@
         "pin": "#ebebeb",
         "satellite": "#ebebeb"
     },
-    "injectCSS": "section#filesystem{left:0;width:100vw}section#filesystem>h3.title,section#filesystem>div{width:100vw}section#keyboard{display:none;}"
+    "injectCSS": "section#filesystem{left: 0;width:82vw} section#filesystem>h3.title, section#filesystem>div{width:82vw;padding-left:10px;} section#keyboard{display:none;}"
```
Comment (GitSquared) [inline src/assets/themes/apollo-notype.json:29]: Same matter as above... Does this break on 16:9 and other common screen ratios?

**Review by GitSquared — CHANGES_REQUESTED** [review]

File: src/assets/css/filesystem.css (hunk 2 above)
Comment (geosp) [reply src/assets/css/filesystem.css:4]: I tested this code in multiple aspects ratios and it works as expected. But I will follow your advice, move my changes to the recommended location and make sure it works.

File: src/assets/themes/apollo-notype.json (hunk 3 above)
Comment (geosp) [reply src/assets/themes/apollo-notype.json:29]: I tested this code in multiple aspects ratios and it works as expected. But I will follow your advice, move my changes to the recommended location and make sure it works.

File: src/assets/css/modal.css (hunk 1 above)
Comment (geosp) [reply src/assets/css/modal.css:135]: Sorry, are you saying there is a conflict? I don't see it.

Comment (geosp) [reply src/assets/css/modal.css:135]: I don't see the conflicts.

**Comment by geosp** [comment]
I would take a substantially different approach in regards to how components are layout and created in this application. At this time the changes made in the initial PR  I submitted work for 4:3 and 16:9 aspect ratios. I tried to accommodate the approach you suggested but is not working for me. I will work in my current fork to experiment with more radical changes including the introduction of react components and a layout manager.  Thank you for the time you took to review my PR. I hope you are able to fix these issues without having to make more radical changes than what I'm suggesting. I really like what you have done here, I'm definitely enjoying using it on a daily basis.

**Comment by GitSquared** [comment]
> I would take a substantially different approach in regards to how components are layout and created in this application.

Yes, I would too, but this was written 4 years ago.

> Thank you for the time you took to review my PR.

Thank you for taking the time to play around with it. Sorry that it didn't work out... maybe next time?

**Comment by geosp** [comment]
tron_4x3

**Comment by geosp** [comment]
Also, I removed this file `src/assets/css/extra_ratios.css`, and seems to have no impact on the layout of components. I'm not sure what is going on but it seems like there is some other more fundamental issue. I really cool that this application is still relevant after four years.

**Comment by GitSquared** [comment]
> and seems to have no impact on the layout of components.

It's just a bunch of CSS overrides for specific screens witdth:height ratios.

**Pull request closed without merging**